src/main/java/org/sunbird/workflow/service/impl/SignUpWFServiceImpl.java

```diff
@@ -0,0 +1,240 @@
+        } else {
+            response.getParams().setStatus(Constants.SUCCESSFUL);
+            response.setResponseCode(HttpStatus.OK);
+        }
+        return response;
+    }
+
+    private String validateWfRequest(WfRequest wfRequest) {
+        List<String> params = new ArrayList<String>();
+        StringBuilder strBuilder = new StringBuilder();
+        if (StringUtils.isEmpty(wfRequest.getUserId())) {
+            params.add(Constants.USER_UUID_VALIDATION_ERROR);
+        }
+        if (StringUtils.isEmpty(wfRequest.getActorUserId())) {
+            params.add(Constants.ACTOR_UUID_VALIDATION_ERROR);
+        }
+        if (StringUtils.isEmpty(wfRequest.getUserId())) {
+            params.add(Constants.APPLICATION_ID_VALIDATION_ERROR);
+        }
+        if (StringUtils.isEmpty(wfRequest.getState())) {
+            params.add(Constants.STATE_VALIDATION_ERROR);
+        }
+        if (CollectionUtils.isEmpty(wfRequest.getUpdateFieldValues())) {
+            params.add(Constants.FIELD_VALUE_VALIDATION_ERROR);
+        }
+        if (StringUtils.isEmpty(wfRequest.getAction())) {
+            params.add(Constants.ACTION_VALIDATION_ERROR);
+        }
+        if (StringUtils.isEmpty(wfRequest.getServiceName())) {
+            params.add(Constants.WORKFLOW_SERVICENAME_VALIDATION_ERROR);
+        }
+        if (!params.isEmpty()) {
+            strBuilder.append("Invalid Request. " + params);
+        }
+        return strBuilder.toString();
+    }
+
+    private SBApiResponse changeStatus(String rootOrg, String org, WfRequest wfRequest, SBApiResponse response) {
+        String wfId = wfRequest.getWfId();
+        String nextState = null;
+        String errMsg = null;
+        try {
+            WorkFlowModel workFlowModel = wfService.getWorkFlowConfig(wfRequest.getServiceName());
+            WfStatus wfStatus = wfService.getWfStatus(wfRequest.getState(), workFlowModel);
+            if (wfStatus == null) {
+                response.getParams().setErrmsg(Constants.WORKFLOW_STATE_CHECK_ERROR);
+                return response;
+            }
+
+            WfAction wfAction = wfService.getWfAction(wfRequest.getAction(), wfStatus);
+            if (wfAction == null) {
+                response.getParams().setErrmsg(Constants.WORKFLOW_ACTION_ERROR);
+                return response;
+            }
+            nextState = wfAction.getNextState();
+
+            WfStatus wfNextStatus = wfService.getWfStatus(nextState, workFlowModel);
+            if (wfNextStatus == null) {
+                response.getParams().setErrmsg(Constants.WORKFLOW_STATE_CHECK_ERROR);
+                return response;
+            }
+            WfStatusEntity applicationStatus = wfStatusRepo.findByApplicationIdAndWfId(wfRequest.getApplicationId(), wfRequest.getWfId());
+            errMsg = validateUserAndWfStatus(wfRequest, wfStatus, applicationStatus);
+            if (StringUtils.isEmpty(errMsg)) {
+                if (ObjectUtils.isEmpty(applicationStatus)) {
```

Comment:
No two if clauses are required.Put them in a single statement

Reply:
two if condition required here , because conditions are different